cpu/stm32/stmclk/stmclk_mp1.c

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (C) 2020 Savoir-faire Linux
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     cpu_stm32
+ * @{
+ *
+ * @file
+ * @brief       Implementation of STM32 clock configuration for STM32MP1
+ *
+ * @author      Gilles DOFFE <gilles.doffe@savoirfairelinux.com>
+ *
+ * @}
+ */
+
+#include "cpu.h"
+#include "stmclk.h"
+#include "periph_conf.h"
+
+#ifdef STM32MP1_ENGINEERING_MODE
+
+/* make sure we have all needed information about the clock configuration */
+#ifndef CLOCK_HSE
+#error "Please provide CLOCK_HSE in your board's perhip_conf.h"
```

Comment:
```suggestion
#error "Please provide CLOCK_HSE in your board's periph_conf.h"
```

Reply:
Regarding this file, in general, that would be super nice if you could follow the current clock initialization design used by other families from the start. Otherwise, someone (probably me ;) ) will have to refactor it after.